core/src/main/java/org/geysermc/geyser/network/ConnectorServerEventHandler.java

```diff
@@ -125,6 +129,36 @@ public BedrockPong onQuery(InetSocketAddress inetSocketAddress) {
             pong.setSubMotd(config.getBedrock().secondaryMotd());
         }
 
+        GeyserBedrockPing[] geyserPing = {
+                new GeyserBedrockPing(pong.getMotd(), pong.getSubMotd(), pong.getPlayerCount(), pong.getMaximumPlayerCount())
+        };
+        this.geyser.eventBus().fire(new GeyserPingEventImpl() {
+            @Override
+            public GeyserBedrockPing getPing() {
+                return geyserPing[0];
+            }
+
+            @Override
+            public GeyserBedrockPing.GeyserPingBuilder getPingAsBuilder() {
+                return getPing().toBuilder();
+            }
+
+            @Override
+            public void setPing(@NonNull GeyserBedrockPing ping) {
+                geyserPing[0] = ping;
+            }
+
+            @Override
+            public void setPing(GeyserBedrockPing.@NotNull GeyserPingBuilder builder) {
+                setPing(builder.build());
+            }
+        });
```

Comment:
This array stuff and anonymous implementation would be cleaned up a lot if `GeyserDefineCommandsEventImpl` was not abstract as explained above.

Also, all of this new logic should be wrapped an if-statement if there are no listeners for the event, so that there are no new allocations if it would be unecessary (as described by Redned)